notifications.bs

```diff
@@ -735,8 +733,32 @@ constructor steps are:
 <h3 id=static-members>Static members</h3>
 
 <p>The static <dfn attribute for=Notification><code>permission</code></dfn> getter steps are to
-return the <a>permission</a> for the <a>current settings object</a>'s
-<a for="environment settings object">origin</a>.
+return the result of <a>querying the notifications permission state</a>.
+
+<p>To <dfn>query the notifications permission state</dfn>, run these steps:
+
+<ol>
+ <li><p>Let <var>settings</var> be the <a>current settings object</a>.
+
+ <li><p>If <var>settings</var> is not a <a>secure context</a>, return "<code>denied</code>".
+
+ <li><p>Otherwise, let <var>permissionDescriptor</var> be the {{PermissionDescriptor}} with
+ {{PermissionDescriptor/name}} set to "<code>notifications</code>".
+
+ <li><p>Let <var>permissionStatus</var> be the result of
+ <a lt="create a PermissionStatus">creating a `PermissionStatus`</a> for
+ <var>permissionDescriptor</var>.
+
+ <li><p>Run the <a for="powerful feature">permission query algorithm</a> with
+ <var>permissionDescriptor</var> and <var>permissionStatus</var>.
+
+ <li><p>Let <var>permissionState</var> be <var>permissionStatus</var>'s {{PermissionStatus/state}}.
+
+ <li><p>If <var>permissionState</var> is {{PermissionState/"prompt"}}, then set
+ <var>permissionState</var> to "<code>default</code>".
```

Comment:
Maybe this should just return "default"?

Reply:
Yeah, good point.

Reply:
This got lost in the rebase mixup I'm afraid.